test/standalone/x86_cpu_model/main.c

```diff
@@ -0,0 +1,19 @@
+#include <stdio.h>
+
+int foo(void) __attribute__ ((__target__("default")));
+int foo(void) __attribute__ ((__target__("avx2")));
+int foo(void) __attribute__ ((__target__("avx512vnni")));
+
+int main() {
+    printf("%d\n", foo());
+}
+
+int __attribute__ ((__target__("default"))) foo(void) {
+    return 1;
+}
+int __attribute__ ((__target__("avx2"))) foo(void) {
+    return 2;
+}
+int __attribute__ ((__target__("avx512vnni"))) foo(void) {
+    return 3;
+}
```

Comment:
nit: standard compliant must include a newline at end of file or might introduce ub, see https://gcc.gnu.org/legacy-ml/gcc/2001-07/msg01120.html

Reply:
Oh wow I did not know that could happen.

https://github.com/ziglang/zig/pull/18193/commits/6d50d8715a46e7ea30aa59a07261067f86390333